### DISCUSSION TITLE
AOC 2022 Day Seven 💻

### DISCUSSION THREAD

**Discussion by bogdaaamn** [Virtual-Coffee/virtualcoffee.io#765] [Advent of Code 2022]
**Discussion of solutions for [Advent of Code Day 7](https://adventofcode.com/2022/day/7) here.**

Don't forget about the VC Advent of Code private leaderboard:
- Go to [leaderboard/private](https://adventofcode.com/2022/leaderboard/private)
- Use join code `363496-e6f42283`

**Comment by rhdeck** [comment]
Like AOC, I'm a tree-hugger: https://github.com/rhdeck/aoc-2022/blob/main/src/7/day_7.ipynb

Not as terse as I'd like, but introduced me to class properties and methods in Python

**Reply by tkshill** [reply]
It's nice to read some full force python. Love the mix of fp and OOP. Giving this a full review to pick up some technique. Great solve!

**Comment by doleraj** [comment]
Hello today I wrote the Dumbest Possible class-based TS because there are ongoing prod issues and I need to be working on that and not this :C

https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day07/index.ts

**Reply by doleraj** [reply]
I guess, to my credit, the solutions for both parts worked on the first try.

**Reply by tkshill** [reply]
@doleraj First try I think that makes you the king of AOC. Love the use of enums.

**Reply by danieltott** [reply]
That's a win in my book! This one broke my record for failed attempts on both parts.

**Reply by danieltott** [reply]
@tkshill I thought we decided `enum`s are evil...at least that's what @JoeKarow said and I trust him

**Reply by doleraj** [reply]
They're not great. They work in this instance.

**Reply by danieltott** [reply]
To be clear, this wasn't me poking fun at your code, just at @tkshill (I used `enum` on day 2) 😁

**Reply by doleraj** [reply]
Yeah, I didn't parse it that way. TS/JS enums are woefully underpowered when compared to other languages.

**Reply by tkshill** [reply]
Yeah, Dan and I spent an hour trying to understand enum interactions with TS. It didn't work.

**Comment by danieltott** [comment]
Gosh was I bad at today. Spent WAY too long trying to be clever and avoid recursion for some reason.

Also I'm bad at reading it turns out.

Solution: https://github.com/danieltott/aoc2022/blob/main/src/day07/index.ts

PR: https://github.com/danieltott/aoc2022/pull/4

**Reply by danieltott** [reply]
Definitely can improve the performance here without too much trouble - don't need that `filter(Boolean)` for instance. 

Have spent too much time on this though 😂

**Reply by doleraj** [reply]
I am jealous of folks who realized they could just ... ignore most of the input and just calculate the sizes directly

**Reply by tkshill** [reply]
I didn't know if I'd need the file details so I kept em. 😭

**Comment by tkshill** [comment]
I love and hate this solve. But this is my favourite type of problem. And I'm happy with lots of this solution, even though it isn't sparse.

I'm with @danieltott , all my problems are reading comprehension

[solve](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day07.fs)

**Comment by turingtestable** [comment]
[Ruby solution](https://github.com/turingtestable/AdventOfCode2022/blob/master/challenge7/solution.rb) I love these recursive tree things, my biggest issue was a sick kid at home taking my attention.
